### ISSUE TITLE
Failed ME Firmware on Lenovo P1 Gen4i

### ISSUE THREAD

**Issue by Jiyangson** [fwupd/firmware-lenovo#332]
I'm having the same problem of failing to recover ME Firmware as  #226. I was using Windows 10 but there was no available solution so I decided to install Ubuntu, and tried to use `fwupdmgr reinstall` or `fwupdmgr --allow-older --allow-reinstall install n35rk02w.cab`. But none of them worked, and it showed no supported devices found
![image](https://user-images.githubusercontent.com/31755447/233791986-e5977611-726d-4a60-8b3b-3debe4b55d9e.png)
It could mean that I need to update the package so I tried `fwupdmgr refresh`, and it showed verification failed
![image](https://user-images.githubusercontent.com/31755447/233792105-03087c90-48d0-4005-9920-8602af05d18b.png)
I'm stuck on this step. What should I do now? Thanks!
Here are some info
```
$ sudo dmidecode | grep SKU
		Corporate SKU
	SKU Number: LENOVO_MT_20Y3_BU_Think_FM_ThinkPad P1 Gen 4i
	SKU Number: Not Specified
```
```
$ fwupdmgr get-devices
20Y3004JUS
│
├─MZVL21T0HCLR-00BL7:
│     Device ID:          04e17fcf7d3de91da49a163ffe4907855c3648be
│     Summary:            NVM Express solid state drive
│     Current version:    CL2QGXA7
│     Vendor:             Samsung (NVME:0x144D)
│     GUIDs:              4d7a2791-106b-5e72-9cfb-8ea3d89f5421
│                         310f81b5-6fce-501e-acfb-487d10501e78
│                         60c89aac-f321-515b-b419-3cf02aa9d375
│                         bec63ed7-a95f-54fe-b8cc-8e9fee64ba5a
│                         c0e40d86-e47a-57fe-8ed1-453e6d83a586
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Signed Payload
│   
├─Prometheus:
│ │   Device ID:          4295296d98b3ba38c72f6baa33d24f03a1d428f6
│ │   Summary:            Fingerprint reader
│ │   Current version:    10.01.3333223
│ │   Vendor:             Synaptics (USB:0x06CB)
│ │   Install Duration:   2 seconds
│ │   GUIDs:              d67c4925-e30f-53e6-abc6-e7af0e4c5884
│ │                       659f7e45-8d45-528d-b3c7-0695eed055f6
│ │   Device Flags:       • Updatable
│ │                       • Cryptographic hash verification is available
│ │                       • Signed Payload
│ │ 
│ └─Prometheus IOTA Config:
│       Device ID:        ea324f9d78d801d381c77ba6947fb8112bebc3a8
│       Summary:          Fingerprint reader config
│       Current version:  0009
│       Minimum Version:  0009
│       Vendor:           Synaptics (USB:0x06CB)
│       GUIDs:            49a675dd-c0d7-5c08-b19a-78fdfb9ea399
│                         d2e397c9-fe28-5506-bf0d-b5860baa33ba
│       Device Flags:     • Updatable
│                         • Only version upgrades are allowed
│                         • Signed Payload
│     
├─System Firmware:
│ │   Device ID:          a45df35ac0e948ee180fe216a5f703f32dda163f
│ │   Summary:            UEFI ESRT device
│ │   Current version:    65558
│ │   Minimum Version:    65536
│ │   Vendor:             LENOVO (DMI:LENOVO)
│ │   Update State:       Success
│ │   GUIDs:              f1918178-1ab6-4d4e-b82c-26e4d0e3869c
│ │                       230c8b18-8d9b-53ec-838b-6cfc0383493a
│ │   Device Flags:       • Internal device
│ │                       • Updatable
│ │                       • System requires external power source
│ │                       • Needs a reboot after installation
│ │                       • Cryptographic hash verification is available
│ │                       • Device is usable for the duration of the update
│ │ 
│ └─UEFI dbx:
│       Device ID:        362301da643102b9f38477387e2193e57abaa590
│       Summary:          UEFI revocation database
│       Current version:  304
│       Minimum Version:  304
│       Vendor:           UEFI:Linux Foundation
│       Install Duration: 1 second
│       GUIDs:            14503b3d-73ce-5d06-8137-77c68972a341
│                         5971a208-da00-5fce-b5f5-1234342f9cf7
│                         c6682ade-b5ec-57c4-b687-676351208742
│                         f8ba2887-9411-5c36-9cee-88995bb39731
│       Device Flags:     • Internal device
│                         • Updatable
│                         • Needs a reboot after installation
│                         • Only version upgrades are allowed
│                         • Signed Payload
│     
├─UEFI Device Firmware:
│     Device ID:          349bb341230b1a86e5effe7dfe4337e1590227bd
│     Summary:            UEFI ESRT device
│     Current version:    4029286587
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               38a375e4-3927-42a2-ab34-e250d748b698
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          2292ae5236790b47884e37cf162dcf23bfcd1c60
│     Summary:            UEFI ESRT device
│     Current version:    65552
│     Minimum Version:    65552
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               ab5a2de4-75ef-4906-9533-143d2acad54e
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          f95c9218acd12697af946874bfe4239587209232
│     Summary:            UEFI ESRT device
│     Current version:    16777223
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               62009ffc-86f3-4a7c-93a6-1b9b0dd6472d
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          d96de5c124b60ed6241ebcb6bb2c839cb5580786
│     Summary:            UEFI ESRT device
│     Current version:    0
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               52c9711e-7a5d-4e8d-ac4a-ba4d920a1008
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          f37fb01122dd62c773f4e84ec89737e059712d59
│     Summary:            UEFI ESRT device
│     Current version:    1
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               f18786ab-c122-4f94-a653-a66a88de271e
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          36efb79c255f402f619fa9eb53cd659db51f2a04
│     Summary:            UEFI ESRT device
│     Current version:    117572096
│     Minimum Version:    117572096
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               25f36ac7-8495-42e7-bd2e-4b35a834ee47
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          11a7fbdd09ce583e58b899660fe10f147fd155a2
│     Summary:            UEFI ESRT device
│     Current version:    2754747
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               072c0411-31b3-4e4f-85bc-1e1b5ed55223
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          a083ebc5138e5e071ef7270cc9a8280722cc7adf
│     Summary:            UEFI ESRT device
│     Current version:    4115
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     GUID:               7c8c8f35-30c9-0032-5e44-5a045511746b
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
└─UEFI Device Firmware:
      Device ID:          5bfc5b91c18ef8b751d3d052af77b69ebf738038
      Summary:            UEFI ESRT device
      Current version:    10551296
      Minimum Version:    57374
      Vendor:             DMI:LENOVO
      Update State:       Success
      GUID:               09f77c9f-1c5d-4616-bafb-bbb19f557480
      Device Flags:       • Internal device
                          • Updatable
                          • System requires external power source
                          • Needs a reboot after installation
                          • Device is usable for the duration of the update
```

**Comment by kmauleon** [comment]
asking ODM... please wait...

**Comment by hughsie** [comment]
Looks like your SSL is failing. Are you using a corporate proxy or behind a firewall perhaps?

**Comment by kmauleon** [comment]
hi @Jiyangson any update from your side please...

**Comment by Jiyangson** [comment]
sorry for the late response. It was the proxy issue and I have fixed it. But new problem occurs. When I updated the system firmware using `fwupdmgr reinstall`, It asked me to restart. After I typed yes, the screen went black and fans were still running for more than an hour. I had to manually turn off and turn back on for three times, and the failed to recover problem was still there. 

BTW This issue has wasted me a lot of time, so is there a way to totally reset my computer?
